### ISSUE TITLE
Cannot create cluster (heketi-cli seems to interpet a redirect as an error?)

### ISSUE THREAD

**Issue by robdewit** [heketi/heketi#663]
I'm running heketi/heketi:latest in docker, executor is 'mock' and authorization is set to false. (Using 'ssh' and auth gives same results).

When I try to setup the cluster with either `heketi-cli topology load --json topology.json` or `heketi-cli cluster create` the response is always:

```
Error:
```
No further messages. The output in the docker server is: 

```
Heketi v3.1.0-2-g900f849-release-3
[heketi] INFO 2017/02/02 14:41:15 Loaded mock executor
[heketi] INFO 2017/02/02 14:41:15 Loaded simple allocator
[heketi] INFO 2017/02/02 14:41:15 GlusterFS Application Loaded
Listening on port 8080
[negroni] Started GET /clusters
[negroni] Completed 200 OK in 88.568µs
[negroni] Started GET /clusters
[negroni] Completed 200 OK in 62.295µs
```

Maybe the following is totally normal, but it's the only thing I found: 

When I strace the heketi-cli program it seems it issues a POST request for `//clusters` with two forward slashes. It receives a 301 advising to request '/clusters', but instead of following the redirect I get the error:

```
[pid  3145] write(3, "POST //clusters HTTP/1.1\r\nHost: localhost:8080\r\nUser-Agent: Go-http-client/1.1\r\nContent-Length: 2\r\nAuthorization: bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE0ODYwNDY4MjYsImlhdCI6MTQ4NjA0NjUyNiwiaXNzIjoiYWRtaW4iLCJxc2giOiJlNDYxYzFjM2Y4ZjZlM2NiNjllYzdiMDI1NTdlODliYTAxZTg1YjU1ODdjNjdlZDhhNTk3MzczNWMzZjVkZGJmIn0.RcaHZu0BIrDiJaBb9nkpw3jXbnFqzWSBAvOiTs_dcT0\r\nContent-Type: application/json\r\nAccept-Encoding: gzip\r\n\r\n{}", 426 <unfinished ...>
[pid  3149] <... read resumed> 0xc4203a5000, 4096) = -1 EAGAIN (Resource temporarily unavailable)
[pid  3149] futex(0xc420062110, FUTEX_WAIT, 0, NULL <unfinished ...>
[pid  3145] <... write resumed> )       = 426
[pid  3145] futex(0xc420062110, FUTEX_WAKE, 1) = 1
[pid  3145] futex(0x12d9410, FUTEX_WAIT, 0, NULL <unfinished ...>
[pid  3149] <... futex resumed> )       = 0
[pid  3149] futex(0xc420062110, FUTEX_WAIT, 0, NULL <unfinished ...>
[pid  3148] <... epoll_wait resumed> {{EPOLLIN|EPOLLOUT, {u32=3281502960, u64=140590446002928}}}, 128, 4294967295) = 1
[pid  3148] read(3, "HTTP/1.1 301 Moved Permanently\r\nLocation: /clusters\r\nDate: Thu, 02 Feb 2017 14:42:06 GMT\r\nContent-Length: 0\r\nContent-Type: text/plain; charset=utf-8\r\n\r\n", 4096) = 152
[pid  3148] futex(0xc420062110, FUTEX_WAKE, 1 <unfinished ...>
[pid  3149] <... futex resumed> )       = 0
[pid  3148] <... futex resumed> )       = 1
[pid  3149] epoll_wait(4,  <unfinished ...>
[pid  3148] read(3,  <unfinished ...>
[pid  3149] <... epoll_wait resumed> {}, 128, 0) = 0
[pid  3148] <... read resumed> 0xc4203a5000, 4096) = -1 EAGAIN (Resource temporarily unavailable)
[pid  3149] select(0, NULL, NULL, NULL, {0, 3} <unfinished ...>
[pid  3148] mmap(NULL, 262144, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fddc392b000
[pid  3148] write(2, "Error: \n", 8Error: 
```

**Comment by lpabon** [comment]
Ah, good point @robdewit . Nice work looking into it. Would you like to send a patch to fix it before I assign it to someone?

**Comment by robdewit** [comment]
I do not have the go-skills to fix it myself. So if someone else would like to pick it up, I'd be more than grateful.

**Comment by petervansickel** [comment]
I ran into similar issue with creating a volume.  I had set the HEKETI_CLI_SERVER to `http://localhost:8080/`.  The error message from heketi-cli was:
`Error: {"volumes":["ccdb21cfd1e83cf9c3299207f66fb705","de785f066ce0e0ce86c39c5fb920682c"]}`.  A lot of other commands were working just fine, e.g., info and list on the various resources.
Also I had the same issue with the resturl in a storageclass definition for kubernetes.  So the persistent volume claim was failing with the same cryptic error message.  
Lesson learned: A trailing slash on the heketi server URL will slash you good. ;-)
